datacube.py

```diff
@@ -0,0 +1,382 @@
+            da_sen1 = da_sen1.drop_vars(names=key)
+
+    # Create xarray.Dataset datacube with VH and VV channels from SAR
+    # da_vh: xr.DataArray = da_sen1.sel(band="vh", drop=True).rename("vh")
+    # da_vv: xr.DataArray = da_sen1.sel(band="vv", drop=True).rename("vv")
+    # ds_sen1: xr.Dataset = xr.merge(objects=[da_vh, da_vv], join="override")
+    da_sen1 = stackstac.mosaic(da_sen1, dim="time")
+
+    # print(ds_sen1)
+
+    # da_sen1 = da_sen1.drop([da_sen1.time[0].values], dim='time') # Remove first scene which has ascending orbit
+
+    da_dem: xr.DataArray = stackstac.stack(
+        items=dem_items,
+        epsg=epsg,
+        bounds_latlon=BBOX,  # W, S, E, N
+        resolution=resolution,
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    da_dem = stackstac.mosaic(da_dem, dim="time")
+
+    # _, index = np.unique(da_dem['time'], return_index=True)  # Remove redundant time
+    # da_dem = da_dem.isel(time=index)
+
+    return da_sen2, da_sen1, da_dem
+
+
+def merge_datarrays(da_sen2, da_sen1, da_dem):
+    """
+    Merge xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+
+    Parameters:
+    - da_sen2 (xr.DataArray): xarray DataArray for Sentinel-2 data.
+    - da_sen1 (xr.DataArray): xarray DataArray for Sentinel-1 data.
+    - da_dem (xr.DataArray): xarray DataArray for DEM data.
+
+    Returns:
+    - xr.DataArray: Merged xarray DataArray.
+    """
+    da_merge = xr.merge([da_sen2, da_sen1, da_dem], compat="override")
+    print("Merged datarray: ", da_merge)
+    print(
+        "Time variables (S2, merged): ", da_sen2.time.values, da_merge.time.values
+    )  # da_sen1.time.values, da_dem.time.values
+    return da_merge
+
+
+def process(year1, year2, aoi, resolution, epsg):
+    """
+    Process Sentinel-2, Sentinel-1, and DEM data for a specified time range, area of interest (AOI),
+    resolution, and EPSG code.
+
+    Parameters:
+    - year1 (int): The starting year of the date range.
+    - year2 (int): The ending year of the date range.
+    - aoi (shapely.geometry.base.BaseGeometry): Geometry object for an Area of Interest (AOI).
+    - resolution (int): Spatial resolution.
+    - epsg (int): EPSG code for the coordinate reference system.
+
+    Returns:
+    - xr.DataArray: Merged xarray DataArray containing processed data.
+    """
+
+    date, YEAR, MONTH, DAY, CLOUD = get_conditions(year1, year2)
+    week = get_week(YEAR, MONTH, DAY)
+
+    catalog, s2_items, s2_items_gdf, BBOX = search_sentinel2(week, aoi)
+
+    s1_items, s1_gdf = search_sentinel1(BBOX, catalog, week)
+    # s1_items, s1_gdf = search_sentinel1_calc_max_area(BBOX, catalog, week) # WIP
+
+    dem_items, dem_gdf = search_dem(BBOX, catalog, week, s2_items)
+
+    da_sen2, da_sen1, da_dem = make_dataarrays(
+        s2_items, s1_items, dem_items, BBOX, resolution, epsg
+    )
+
+    da_merge = merge_datarrays(da_sen2, da_sen1, da_dem)
```

Comment:
As a sanity check, could you plot each invidual band from this datacube and see if they look ok? I've had issues with striped outputs when running `xr.merge` before on Sentinel-1 and Copernicus DEM, and just want to double check.

Reply:
Should be addressed in https://github.com/Clay-foundation/model/pull/27#issuecomment-1811506452